Add bulk results section (#154)
AlexsLemonade/ScPCA-manuscript: content/03.results.md

If the submitter provided cell types, the submitter annotations are compared to the annotations from both `SingleR` and `CellAssign`.
A summary of this comparison is included in the cell type report along with a table summarizing the submitter cell type annotations and a UMAP plot where each cell is colored by the submitter annotation.
The Jaccard similarity index is calculated for all pairs of cell type labels in submitter annotations and `SingleR` annotations and in submitter annotations and `CellAssign` annotations.

> The results from both comparisons are displayed in a stacked heatmap available in the report, an example of which is shown in Figure {@fig:figS7}.

Reviewer: ```suggestion
The results from both comparisons are displayed in a stacked heatmap available in the report, an example of which is shown in Figure {@fig:figS5}B.
```
This shouldn't have been changed.

We observed an expected positive relationship between bulk and pseudobulk expression across all projects.

We next performed an overrepresentation analysis to test if cell type marker genes for our consensus cell types were enriched in either modality, which may in turn indicate that certain cell types were more prevalently detected by either modality.
To this end, we calculated the per-gene median of each project's model residuals and identified outliers, where positive outliers represent genes that are enriched in bulk RNA-seq, and negative outliers represent genes that are enriched in single-cell/nuclei RNA-seq.
For each modality, we calculated the odds ratio as: the odds a cell type marker gene is present in the given modality's outliers compared to other genes.
Following permutation testing and P-value correction to control the FDR at $5%$, we indeed found that several cell type marker gene sets were overrepresented in the bulk modality, but never in the pseudobulk modality (Figure {@fig:fig6}B, Figure {@fig:figS8}B).

> In brain and CNS tumors, we nearly exclusively identified stromal (e.g., endothelial and extracellular matrix secreting cells) and/or neuronal cell types (e.g., glial cells and astrocytes), all of which are known to be prevalent non-immune cells in glioma tumor microenvironments [@doi:10.3389/fimmu.2023.1227126; @doi:10.3389/fphar.2024.1355242], as overrepresented in bulk as opposed to immune cell types (Figure {@fig:fig6}B).

Reviewer: ```suggestion
In brain and CNS tumors, we nearly exclusively identified stromal (e.g., endothelial and extracellular matrix secreting cells) and/or neuronal cell types (e.g., glial cells and astrocytes), all of which are known to be prevalent non-immune cells in glioma tumor microenvironments [@doi:10.3389/fimmu.2023.1227126; @doi:10.3389/fphar.2024.1355242], as overrepresented in bulk (Figure {@fig:fig6}B).
```
I don't think you need the phrase about immune cell types.

This difference may reflect that single-cell approaches have increased sensitivity to detect immune cells relative to single-nuclei approaches [@doi:10.4132/jptm.2022.12.19].

Given that our consensus cell type analysis identified a variety of immune cells from high- and low-grade gliomas (Figure {@fig:fig5}), this analysis suggests that non-immune cells may have been lost during single-cell library preparation.
Indeed, several of these overrepresented bulk cell types for `SCPCP000001` and `SCPCP000002` were not among the single-nuclei consensus cell types themselves (`SCPCP000001`: "blood vessel endothelial cell", "extracellular matrix secreting cell", "glial cell", "pericyte"; `SCPCP000002`: "blood vessel endothelial cell", "extracellular matrix secreting cell", "microvascular endothelial cell"), further emphasizing loss of these cell types in the single-nuclei data.
By contrast, we uncovered a variety of both immune and non-immune cell types overrepresented in bulk RNA-seq `SCPCP000017` (osteosarcoma; Figure {@fig:figS8}B), which may reflect inherent challenges in dissociating bone tissue [@doi: https://doi.org/10.1186/s12885-023-10977-1].

> Taken together, these results show that, while bulk and single-cell/nuclei expression is indeed highly correlated, differences in cell type composition may still be present between modalities potentially driven by cell-type--specific loss in single-cell experiments.

Reviewer: ```suggestion
Taken together, these results show that, while bulk and single-cell/nuclei expression is indeed highly correlated, differences in cell type composition may still be present between modalities potentially driven by cell-type-specific loss in single-cell experiments.
```

Previous research has suggested that, compared to bulk RNA-seq, and single-cell RNA-seq may fail to capture certain cell types [@doi: @10.1038/s41587-020-0465-8] due, for example, to technical aspects of library preparation.
We therefore sought to ask whether we could identify differences in cell type composition between these two modalities.
For this analysis, we specifically focused on ScPCA projects with solid tumors, considering only non-multiplexed samples with both sequencing modalities and excluding low-quality single-cell libraries.
In total, we analyzed 97 samples across five projects (`SCPCP00001`, `SCPCP000002`, `SCPCP000006`, `SCPCP000009`, and `SCPCP000017`), all of which were single-nuclei experiments.
As described in Materials and Methods, we derived pseudobulk expression matrices from the single-nuclei libraries, and we compared their expression to bulk using a series of linear models (one per ScPCA project) predicting bulk from pseudobulk expression, with a random effect controlling for sample (Figure {@fig:fig6}A, Figure {@fig:figS8}A).
We observed an expected positive relationship between bulk and pseudobulk expression across all projects.

> We next performed an overrepresentation analysis to test if cell type marker genes were enriched either modality.

Reviewer: ```suggestion
We next performed an overrepresentation analysis to test if cell type marker genes were enriched in either modality.
```

We calculated the per-gene median of each project's model residuals and identified outliers in the bulk ($>2.5\sigma$) or pseudobulk direction ($<-2.5\sigma$).
Using a set of cell type marker genes for each consensus cell type, we calculated the odds ratio for each modality as: the odds a cell type marker gene is present in the given modality's outliers compared to other genes.
Following permutation testing and P-value correction to control the FDR at $5%$, we indeed found that several cell type marker gene sets were overrepresented in the bulk modality, but never in the pseudobulk modality (Figure {@fig:fig6}B, Figure {@fig:figS8}B).
This in turn suggests that those cell types may be more prevalently detected in bulk RNA-Seq.

> Most notably, for projects `SCPCP000001` (high-grade gliomas) and `SCPCP000002` (low-grade gliomas), several of these overrepresented cell types were not among the single-nuclei consensus cell types themselves (`SCPCP000001`: "blood vessel endothelial cell", "extracellular matrix secreting cell", "glial cell", "pericyte"; `SCPCP000002`: "blood vessel endothelial cell", "extracellular matrix secreting cell", "microvascular endothelial cell"), many of which are expected to be found in the glioma tumor microenvironment [doi: @10.3389/fimmu.2023.1227126].

Reviewer: I think I would actually cite this paper which specifically talks about the neural component since the majority of these cell types are either neural or stromal, not immune. We don't have a problem identifying immune cells, but it appears that we are missing identificaiton of neural cell types in the single-cell. 

https://academic.oup.com/noa/article/5/1/vdad009/7053170

This paper also has a section for non-immune cells. 
https://www.frontiersin.org/journals/pharmacology/articles/10.3389/fphar.2024.1355242/full

Reviewer: Also, I think we want to first discuss more generally all of the cell types that are over represented in bulk as compared to pseudobulk. I would mention things like neurons, astrocytes, and endothelial cells and state that although we see them in pseudobulk (see consensus annotation figure), they seem to be over represented in bulk, meaning that we might be losing some of these cell types when performing single nuclei and enriching for immune cell types over other normal cell types we may see in bulk. 

I think you can keep the sentence that you have currently about those that are found in bulk but not detected in pseudobulk, but I think that should be secondary after you discuss the main overall finding with all the cell types that are over represented.

A summary of this comparison is included in the cell type report along with a table summarizing the submitter cell type annotations and a UMAP plot where each cell is colored by the submitter annotation.
The Jaccard similarity index is calculated for all pairs of cell type labels in submitter annotations and `SingleR` annotations and in submitter annotations and `CellAssign` annotations.
The results from both comparisons are displayed in a stacked heatmap available in the report, an example of which is shown in Figure {@fig:figS5}B.

## Analysis of bulk RNA-seq

Several projects in the ScPCA Portal quantified many samples' expression both at the bulk and single-cell level.
Previous research has suggested that, compared to bulk RNA-seq, single-cell RNA-seq may fail to capture certain cell types [@doi:10.1038/s41587-020-0465-8] due, for example, to technical aspects of library preparation.

> We therefore sought to ask whether we could identify differences in cell type composition between these two modalities.

Reviewer 2: I would maybe soften this a little -- we didn't really directly ask about cell type composition with this analysis.

In total, we analyzed 97 samples across five projects: `SCPCP00001`, `SCPCP000002`, `SCPCP000006`, `SCPCP000009`, and `SCPCP000017`.
Projects `SCPCP000001` and `SCPCP000002` comprise high- and low-grade gliomas, respectively, and were sequenced at the bulk and single-cell levels, and `SCPCP000006`, `SCPCP000009`, and `SCPCP000017` comprise Wilms tumors, CNS tumors, and osteosarcomas, respectively, and were sequenced at the bulk and single-nuclei levels.
As described in Materials and Methods, we derived pseudobulk expression matrices from the single-nuclei libraries, and we compared their expression to bulk using a series of linear models (one per ScPCA project) predicting bulk from pseudobulk expression, with a random effect controlling for sample (Figure {@fig:fig6}A, Figure {@fig:figS8}A).
We observed an expected positive relationship between bulk and pseudobulk expression across all projects.

We next performed an overrepresentation analysis to test if cell type marker genes for our consensus cell types were enriched in either modality, which may in turn indicate that certain cell types were more prevalently detected by either modality.

> To this end, we calculated the per-gene median of each project's model residuals and identified outliers, where positive outliers represent genes that are enriched in bulk RNA-seq, and negative outliers represent genes that are enriched in single-cell/nuclei RNA-seq.

Reviewer 2: I'm not sure "enriched" is the right word here. Positive outliers are genes with higher expression in bulk RNA-seq than predicted based on the pseudobulk expression, and negative outliers are genes with lower expression in bulk RNA-seq than predicted on the pseudobulk expression.

Author: > I'm not sure "enriched" is the right word here

I agree very much, in particular since it implies a different type of analysis to, but I wasn't landing on different verbiage. Will rework with some of this suggested phrasing

In brain and CNS tumors, we nearly exclusively identified stromal (e.g., endothelial and extracellular matrix secreting cells) and/or neuronal cell types (e.g., glial cells and astrocytes), all of which are known to be prevalent non-immune cells in glioma tumor microenvironments [@doi:10.3389/fimmu.2023.1227126; @doi:10.3389/fphar.2024.1355242], as overrepresented in bulk (Figure {@fig:fig6}B).
The only exception to this was monocytes being overrepresented in bulk for `SCPCP000009` (brain and CNS tumors), which was sequenced at the single-nuclei level while projects `SCPCP000001` (high-grade gliomas) and `SCPCP000002` (low-grade gliomas) were sequenced at the single-cell level.
This difference may reflect that single-cell approaches have increased sensitivity to detect immune cells relative to single-nuclei approaches [@doi:10.4132/jptm.2022.12.19].

Given that our consensus cell type analysis identified a variety of immune cells from high- and low-grade gliomas (Figure {@fig:fig5}), this analysis suggests that non-immune cells may have been lost during single-cell library preparation.

> Indeed, several of these overrepresented bulk cell types for `SCPCP000001` and `SCPCP000002` were not among the single-nuclei consensus cell types themselves (`SCPCP000001`: "blood vessel endothelial cell", "extracellular matrix secreting cell", "glial cell", "pericyte"; `SCPCP000002`: "blood vessel endothelial cell", "extracellular matrix secreting cell", "microvascular endothelial cell"), further emphasizing loss of these cell types in the single-nuclei data.

Reviewer 2: Aren't these projects single-cell?
```suggestion
Indeed, several of these overrepresented bulk cell types for `SCPCP000001` and `SCPCP000002` were not among the single-cell consensus cell types themselves (`SCPCP000001`: "blood vessel endothelial cell", "extracellular matrix secreting cell", "glial cell", "pericyte"; `SCPCP000002`: "blood vessel endothelial cell", "extracellular matrix secreting cell", "microvascular endothelial cell"), further emphasizing the potential loss of these cell types in the single-cell data.
```

Author: Oooof yes rough typo. I will generally read over to check if this issue appears elsewhere.

> By contrast, we uncovered a variety of both immune and non-immune cell types overrepresented in bulk RNA-seq `SCPCP000017` (osteosarcoma; Figure {@fig:figS8}B), which may reflect inherent challenges in dissociating bone tissue [@doi: https://doi.org/10.1186/s12885-023-10977-1].

Reviewer 2: ```suggestion
By contrast, we uncovered a variety of both immune and non-immune cell types overrepresented in bulk RNA-seq `SCPCP000017` (osteosarcoma; Figure {@fig:figS8}B), which may reflect inherent challenges in dissociating bone tissue [@doi:10.1186/s12885-023-10977-1].
```
